src/components/Tooltip/component.tsx

```diff
@@ -1,7 +1,108 @@
-import React from "react";
+import {useState, useRef, useEffect, ReactNode} from "react";
+import {motion, AnimatePresence} from "framer-motion"
 
-const Tooltip = () => {
-  return <div>Tooltip</div>;
+//content
+//customizable positions
+//triggers on button hover
+//delays
+
+interface TooltipProps {
+  children: ReactNode;
+  content: string;
+  position: 'top' | 'bottom' | 'left' | 'right';
+}
+
+function Tooltip({children, content, position}: TooltipProps) {
+  const [isVisible, setIsVisible] = useState(false)
+  const [shouldRender, setShouldRender] = useState(false)
+  const wrapperRef = useRef<HTMLDivElement | null> (null)
+
+  useEffect(() => {
+
+    const triggerElement = wrapperRef.current
+  
+    const showTooltip = () => {
+      setShouldRender(true)
+      setTimeout(()=> 
+        setIsVisible(true), 300)
+    };
+
+    const hideTooltip = () => {
+      setIsVisible(false)
+      setTimeout(()=> setShouldRender(false), 200) 
+    };
+  
+    if (triggerElement) {
+      // hover start
+      triggerElement.addEventListener('mouseenter', showTooltip);
```

Comment:
instead of using ref + useEffect, I believe you could use onMouseEnter, onMouseLeave, onFocus, onBlur